save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/TestSuitesSourceController.kt

```diff
@@ -328,6 +314,41 @@ class TestSuitesSourceController(
         }
         .map { it.toDto() }
 
+    /**
+     * @param executionId
+     * @return selected [TestSuitesSourceDto] with versions for [com.saveourtool.save.entities.Execution] found by provided values
+     */
+    @PostMapping("/download-snapshot-by-execution-id", produces = [MediaType.APPLICATION_OCTET_STREAM_VALUE])
+    @RequiresAuthorizationSourceHeader
+    @PreAuthorize("permitAll()")
+    @Operation(
+        method = "POST",
+        summary = "Download a snapshot of test suites source which is assigned to execution.",
+        description = "Download a snapshot of test suites source which is assigned to execution.",
+    )
+    @Parameters(
+        Parameter(name = "executionId", `in` = ParameterIn.QUERY, description = "ID of execution", required = true),
+    )
+    @ApiResponse(responseCode = "200", description = "Successfully downloaded snapshot of test suites source for requested execution.")
+    @ApiResponse(responseCode = "404", description = "Execution was not found by provided ID.")
+    @ApiResponse(responseCode = "409", description = "IDs of test suites were not set on requested execution or weren't found by ID.")
+    @ApiResponse(responseCode = "404", description = "Test suites source with such name in organization name was not found.")
+    @ApiResponse(responseCode = "409", description = "Organization was not found by provided name.")
+    fun downloadByExecutionId(
+        @RequestParam executionId: Long
+    ): Mono<ByteBufferFluxResponse> = blockingToMono {
+        val execution = executionService.findExecution(executionId)
+            .orNotFound { "Execution (id=$executionId) not found" }
+        val testSuiteId = execution.parseAndGetTestSuiteIds()?.firstOrNull()
+            .orConflict { "Execution (id=$executionId) doesn't contain testSuiteIds" }
```

Comment:
Why `conflict` and not `not found`?

Reply:
I think it's a conflict situation, when we request download snapshot from execution which doesn't contain testSuiteIds. It means that we request this endpoint before execution is properly populated